[Bug]: Experiment to build with GPU support in WSL2 with RTX 1050

### Issue Type

Build/Install

### Modules Involved

Others

### Have you reproduced the bug with SPU HEAD?

Yes

### Have you searched existing issues?

Yes

### SPU Version

from latest source code

### OS Platform and Distribution

 5.10.102.1-microsoft-standard-WSL2

### Python Version

3.10.13

### Compiler Version

GCC 11.4.0

### Current Behavior?

本机环境：
```bash
(base) limingbo@DESKTOP-mingbo:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
(base) limingbo@DESKTOP-mingbo:~$ uname -a
Linux DESKTOP-mingbo 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
(base) limingbo@DESKTOP-mingbo:~$ nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
(base) limingbo@DESKTOP-mingbo:~$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
(base) limingbo@DESKTOP-mingbo:~$ nvidia-smi
Mon Feb 26 14:56:53 2024
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.65       Driver Version: 527.37       CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0 Off |                  N/A |
| N/A   42C    P8    N/A /  N/A |      0MiB /  4096MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
(base) limingbo@DESKTOP-mingbo:~$ conda activate spu
(spu) limingbo@DESKTOP-mingbo:~$ python
Python 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
(spu) limingbo@DESKTOP-mingbo:~$ gcc --version
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

(spu) limingbo@DESKTOP-mingbo:~$
```
参考[build-with-gpu-support](https://github.com/secretflow/spu/blob/main/INSTALLATION.md#build-with-gpu-support) 试图在WSL2实现目前SPU在GPU方面的兼容测试。
获得报错：


### Standalone code to reproduce the issue

```Python
export ENABLE_GPU_BUILD=1 && python setup.py bdist_wheel
```


### Relevant log output

```Shell
666 |     for (int64_t i = 0; i < sizes_.size; ++i) {
      |                         ~~^~~~~~~~~~~~~
ERROR: /home/limingbo/spu/libspu/cuda_support/BUILD.bazel:32:13: Compiling libspu/cuda_support/utils.cu failed: (Exit 1): nvcc failed: error executing command (from target //libspu/cuda_support:utils)
  (cd /home/limingbo/.cache/bazel/_bazel_limingbo/b283a70469641d1121054548cb95c882/sandbox/linux-sandbox/2392/execroot/spulib && \
  exec env - \
    PATH=/usr/bin \
  /usr/local/cuda-11.8/bin/nvcc -x cu -gencode 'arch=compute_80,code=compute_80' -Xcompiler -fPIC -ccbin /usr/bin/gcc -I . -I bazel-out/k8-opt/bin -I . -I bazel-out/k8-opt/bin -I external/com_google_protobuf -I bazel-out/k8-opt/bin/external/com_google_protobuf -I external/yacl -I bazel-out/k8-opt/bin/external/yacl -I external/com_google_absl -I bazel-out/k8-opt/bin/external/com_google_absl -I external/llvm-project -I bazel-out/k8-opt/bin/external/llvm-project -I external/local_cuda -I bazel-out/k8-opt/bin/external/local_cuda -isystem external/com_google_protobuf/src -isystem bazel-out/k8-opt/bin/external/com_google_protobuf/src -isystem bazel-out/k8-opt/bin/external/zlib/zlib/include -isystem bazel-out/k8-opt/bin/external/com_github_fmtlib_fmt/fmtlib/include -isystem bazel-out/k8-opt/bin/external/com_github_gabime_spdlog/spdlog/include -isystem external/llvm-project/llvm/include -isystem bazel-out/k8-opt/bin/external/llvm-project/llvm/include -isystem external/local_cuda/cuda/include -isystem bazel-out/k8-opt/bin/external/local_cuda/cuda/include -DFMT_HEADER_ONLY -DSPDLOG_FMT_EXTERNAL -DSPDLOG_NO_TLS '-DLLVM_ON_UNIX=1' '-DHAVE_BACKTRACE=1' '-DBACKTRACE_HEADER=<execinfo.h>' '-DLTDL_SHLIB_EXT=".so"' '-DLLVM_PLUGIN_EXT=".so"' '-DLLVM_ENABLE_THREADS=1' '-DHAVE_DEREGISTER_FRAME=1' '-DHAVE_LIBPTHREAD=1' '-DHAVE_PTHREAD_GETNAME_NP=1' '-DHAVE_PTHREAD_H=1' '-DHAVE_PTHREAD_SETNAME_NP=1' '-DHAVE_REGISTER_FRAME=1' '-DHAVE_SETENV_R=1' '-DHAVE_STRERROR_R=1' '-DHAVE_SYSEXITS_H=1' '-DHAVE_UNISTD_H=1' -D_GNU_SOURCE '-DHAVE_LINK_H=1' '-DHAVE_MALLINFO=1' '-DHAVE_SBRK=1' '-DHAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1' -DHAVE_BUILTIN_THREAD_POINTER '-DLLVM_NATIVE_ARCH="X86"' '-DLLVM_NATIVE_ASMPARSER=LLVMInitializeX86AsmParser' '-DLLVM_NATIVE_ASMPRINTER=LLVMInitializeX86AsmPrinter' '-DLLVM_NATIVE_DISASSEMBLER=LLVMInitializeX86Disassembler' '-DLLVM_NATIVE_TARGET=LLVMInitializeX86Target' '-DLLVM_NATIVE_TARGETINFO=LLVMInitializeX86TargetInfo' '-DLLVM_NATIVE_TARGETMC=LLVMInitializeX86TargetMC' '-DLLVM_NATIVE_TARGETMCA=LLVMInitializeX86TargetMCA' '-DLLVM_HOST_TRIPLE="x86_64-unknown-linux-gnu"' '-DLLVM_DEFAULT_TARGET_TRIPLE="x86_64-unknown-linux-gnu"' '-DLLVM_VERSION_MAJOR=19' '-DLLVM_VERSION_MINOR=0' '-DLLVM_VERSION_PATCH=0' '-DLLVM_VERSION_STRING="19.0.0git"' -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS '-DBLAKE3_USE_NEON=0' -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512 -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 --dopt on -Xcompiler -g0 -O2 -DNDEBUG -Xcompiler -ffunction-sections -Xcompiler -fdata-sections -c libspu/cuda_support/utils.cu -o bazel-out/k8-opt/bin/libspu/cuda_support/_objs/utils/utils.pic.o --expt-relaxed-constexpr --extended-lambda)
# Configuration: baf8f5dc15b03ae3d2a3c9acd8354292b899c054e27501fbd3c887951abdd184
# Execution platform: @local_execution_config_platform//:platform

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
external/com_google_absl/absl/strings/str_cat.h: In function 'std::enable_if_t<(sizeof... (T) > 1), typename std::common_type<typename std::conditional<true, void, absl::lts_20240116::strings_internal::EnableIfFastCase<T, typename std::enable_if<((std::is_arithmetic<T>::value && (! std::is_same<T, char>::value)) && (! std::is_same<T, bool>::value)), void>::type> >::type ...>::type> absl::lts_20240116::StrAppend(absl::Nonnull<int_type*>, T ...)':
external/com_google_absl/absl/strings/str_cat.h:606:316: error: expected ';' before '...' token
  606 |   for (const SomeTrivialEmptyType& dummy2 :
      |

           ^
      |

           ;
external/com_google_absl/absl/strings/str_cat.h:606:43: error: parameter packs not expanded with '...':
  606 |   for (const SomeTrivialEmptyType& dummy2 :
      |                                           ^


external/com_google_absl/absl/strings/str_cat.h:606:43: note:         'args'
INFO: Elapsed time: 1091.445s, Critical Path: 572.38s
INFO: 3259 processes: 865 internal, 2394 linux-sandbox.
FAILED: Build did NOT complete successfully
Traceback (most recent call last):
  File "/home/limingbo/spu/setup.py", line 266, in <module>
    setuptools.setup(
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup
    return distutils.core.setup(**attrs)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/dist.py", line 989, in run_command
    super().run_command(command)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 364, in run
    self.run_command("build")
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/dist.py", line 989, in run_command
    super().run_command(command)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 131, in run
    self.run_command(cmd_name)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/dist.py", line 989, in run_command
    super().run_command(command)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/home/limingbo/spu/setup.py", line 237, in run
    return pip_run(self)
  File "/home/limingbo/spu/setup.py", line 208, in pip_run
    build(True, True)
  File "/home/limingbo/spu/setup.py", line 164, in build
    return bazel_invoke(
  File "/home/limingbo/spu/setup.py", line 123, in bazel_invoke
    result = invoker(['bazel'] + cmdline, *args, **kwargs)
  File "/home/limingbo/anaconda3/envs/spu/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['bazel', 'build', '--verbose_failures', '-c', 'opt', '--config=gpu', '--config=avx', '--', '//spu:init', '//spu/utils:distributed', '//spu:api']' returned non-zero exit status 1.
(spu) limingbo@DESKTOP-mingbo:~/spu$
```

--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -40,10 +40,10 @@ def _yacl():
         http_archive,
         name = "yacl",
         urls = [
-            "https://github.com/secretflow/yacl/archive/refs/tags/0.4.3b0.tar.gz",
+            "https://github.com/secretflow/yacl/archive/refs/tags/0.4.3b1.tar.gz",
         ],
-        strip_prefix = "yacl-0.4.3b0",
-        sha256 = "76f40897a360c9f8cce25840d6a068144cc703c630e0a23679f7d2aae3a015d1",
+        strip_prefix = "yacl-0.4.3b1",
+        sha256 = "e22a1ca88a97d4709fb08051d4f756c3bb581ff74f93e01f9eb4bb52dd694b19",
     )
 
 def _libpsi():

